app/assets/javascripts/Components/markdown_editor.jsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import MarkdownPreview from "./markdown_preview";
+import {Tab, Tabs, TabList, TabPanel} from "react-tabs";
+
+import PropTypes from "prop-types";
+
+export default class MarkdownEditor extends React.Component {
+  render() {
+    return (
+      <Tabs disableUpDownKeys>
+        <TabList>
+          <Tab>{I18n.t("write")}</Tab>
+          <Tab id="preview-tab">{I18n.t("preview")}</Tab>
+        </TabList>
+        <TabPanel forceRender>
+          <label>
+            <textarea
+              required={true}
+              id={this.props.text_area_id}
+              name="content"
+              placeholder={I18n.t("results.annotation.placeholder")}
+              value={this.props.content}
+              onChange={this.props.handleChange}
+              rows="8"
+              autoFocus={true}
+            />
+          </label>
+
+          <div className={this.props.show_autocomplete ? "" : "hidden"}>
+            <ul className="tags" key="auto_completion_category">
+              <li className="annotation_category">
+                <p id={this.props.auto_completion_text_id}></p>
+                <div>
+                  <ul id={this.props.auto_completion_list_id}></ul>
+                </div>
+              </li>
+            </ul>
+          </div>
+        </TabPanel>
+        <TabPanel>
+          <MarkdownPreview
+            id="markdown-preview"
+            content={this.props.content}
+            updateAnnotationCompletion={this.props.updateAnnotationCompletion}
+          />
+        </TabPanel>
+      </Tabs>
+    );
+  }
+}
+
+MarkdownEditor.propTypes = {
+  annotation_text_id: PropTypes.string,
```

Comment:
You should remove this prop altogether, it's not being used anywhere. (And don't forget to check the places this component is being used.)